.github/workflows/check.yaml

```
@@ -38,12 +39,17 @@ jobs:
   # This job runs our tests like external parties such as packagers.
   external:
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python: ["3.9", "3.10", "3.11", "3.12", "3.13", "3.14"]
```

I'm not sure it's necessary to test as external packagers do for every version of python. This test is mainly to make sure we're not accidentally on things in `dev-extras` or the like. What is the goal of adding five more tests here?

---

The warning is just saying that it's falling back to a default, which is fine. But even if it makes sense to run tests on `pip install`ed josepy for all python versions, this PR is not the place to do it. If the warning is an issue, this change will keep things as they were before, but without the warning. In which case we also do not need `allow-prereleases: true` added below.

```suggestion
        python: ["3.13"]
```